include/onnxruntime/core/framework/sparse_tensor.h

```diff
@@ -1,72 +1,317 @@
+  /// <returns></returns>
+  SparseFormatFlags FormatFlags() const noexcept {
+    return format_flags_;
   }
 
-  Tensor& MutableValues() {
-    return values_;
+  /// <summary>
+  /// Tests if a specific flag set in the mask
+  /// </summary>
+  /// <param name="flag">flag to be tested</param>
+  /// <returns></returns>
+  bool IsFormatFlagSet(SparseFormatFlags flag) const noexcept {
+    return IsSet(format_flags_, flag);
   }
 
-  Tensor& MutableIndices() {
-    return indices_;
+  /// <summary>
+  /// Returns a would be dense_shape
+  /// This does not describe shapes of values of indices
+  /// and has no relation to
+  /// </summary>
+  /// <returns></returns>
+  const TensorShape& Shape() const noexcept {
+    return dense_shape_;
   }
 
-  //TensorShape& MutableShape() {
-  //  return shape_;
-  //}
+  /// <summary>
+  /// Calculates and returns how much this fully initialized SparseTensor data (would)
+  /// occupy in a contiguous allocation block, or, in fact, occupies if it owns the buffer.
+  /// </summary>
+  /// <returns></returns>
+  int64_t RequiredAllocationSize() const noexcept {
+    return rep_->RequiredAllocationSize();
+  }
+
+  /// <summary>
+  /// Returns Tensor element type enum.
+  /// Useful for type dispatching
+  /// </summary>
+  /// <returns></returns>
+  int32_t GetElementType() const {
+    return ml_data_type_->GetDataType();
+  }
+
+  /// <summary>
+  /// Return Element MLDataType
+  /// </summary>
+  /// <returns></returns>
+  MLDataType DataType() const noexcept {
+    return ml_data_type_;
+  }
+
+  /// <summary>
+  /// Test for string type
+  /// </summary>
+  /// <typeparam name="Rep"></typeparam>
+  /// <returns>true if tensor values are strings</returns>
+  bool IsDataTypeString() const {
+    return utils::IsPrimitiveDataType<std::string>(ml_data_type_);
+  }
+
+  /// <summary>
+  /// Checks if the Tensor contains data type T
+  /// </summary>
+  /// <typeparam name="Builder"></typeparam>
+  /// <returns>true if tensor contains</returns>
+  template <class T>
+  bool IsDataType() const {
+    return utils::IsPrimitiveDataType<T>(ml_data_type_);
+  }
+
+  /// <summary>
+  /// GetRep with format checked. Specialization comes with the actual
+  /// Rep implementation. Any existing rep is destroyed and replaced.
+  /// Specialize for each for the Reps when implemented.
+  /// </summary>
+  /// <typeparam name="Rep"></typeparam>
+  /// <returns>Typed sparse format representation</returns>
+  template <typename Rep>
+  const Rep* GetRep() const;
+
+  /// <summary>
+  /// Create or replace existing mutable representation and make
+  /// available for modification. Specialization comes with the actual
+  /// Rep implementation. The rep is not created at the time of the call
+  /// as some of the necessary data might be missing. Instead, this function
+  /// returns a builder object that allows you call Create() with a rep specific
+  /// argument set.
+  /// </summary>
+  /// <param name="dense shape"></param>
```

Comment:
><param name="dense shape"></param> [](http://example.com/codeflow?start=6&length=34)

general comment: double check documented parameter names
btw, I do appreciate the documentation :)

Reply:
Will do